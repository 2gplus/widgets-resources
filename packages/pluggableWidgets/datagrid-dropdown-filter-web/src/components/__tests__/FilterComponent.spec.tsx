// import { shallow } from "enzyme";
// import { createElement } from "react";
// import { FilterComponent } from "../FilterComponent";
//
// jest.useFakeTimers();
//
// describe("Filter selector", () => {
//     it("renders correctly", () => {
//         const component = shallow(
//             <FilterComponent filterDispatcher={jest.fn()} />
//         );
//
//         expect(component).toMatchSnapshot();
//     });
//
//     it("renders correctly with ariaLabel", () => {
//         const component = shallow(
//             <FilterComponent ariaLabel="my label" filterDispatcher={jest.fn()} />
//         );
//
//         expect(component).toMatchSnapshot();
//     });
//
//     it("calls filterDispatcher when value changes", () => {
//         const filterDispatcher = jest.fn();
//         const component = shallow(
//             <FilterComponent filterDispatcher={filterDispatcher} />
//         );
//
//         const input = component.find("input");
//         input.simulate("change", { target: { value: "test" } });
//
//         expect(filterDispatcher).toBeCalled();
//     });
//
//     it("debounces calls for filterDispatcher when value changes", () => {
//         const filterDispatcher = jest.fn();
//         const component = shallow(
//             <FilterComponent filterDispatcher={filterDispatcher} />
//         );
//
//         // Initial call with default filter
//         expect(filterDispatcher).toBeCalledTimes(1);
//
//         const input = component.find("input");
//         input.simulate("change", { target: { value: "test" } });
//         jest.advanceTimersByTime(499);
//         input.simulate("change", { target: { value: "test2" } });
//         input.simulate("change", { target: { value: "test3" } });
//         jest.advanceTimersByTime(500);
//
//         expect(filterDispatcher).toBeCalledTimes(2);
//
//         input.simulate("change", { target: { value: "test" } });
//         jest.advanceTimersByTime(500);
//
//         expect(filterDispatcher).toBeCalledTimes(3);
//     });
// });
