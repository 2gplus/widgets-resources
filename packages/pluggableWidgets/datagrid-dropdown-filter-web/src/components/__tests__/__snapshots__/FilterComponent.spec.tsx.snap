// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter selector with multi selection renders correctly with ariaLabel 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    aria-label="my label"
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className=""
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_0"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_0"
      >
        1
      </label>
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_1"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_1"
      >
        2
      </label>
    </li>
    <li
      className=""
      key="2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_2"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_2"
      >
        3
      </label>
    </li>
  </ul>
</div>
`;

exports[`Filter selector with multi selection renders correctly with emptyOptioncaption 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value="find me"
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className=""
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_0"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_0"
      >
        1
      </label>
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_1"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_1"
      >
        2
      </label>
    </li>
    <li
      className=""
      key="2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_2"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_2"
      >
        3
      </label>
    </li>
  </ul>
</div>
`;

exports[`Filter selector with multi selection renders correctly with no options 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  />
</div>
`;

exports[`Filter selector with multi selection renders correctly with options 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className=""
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_0"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_0"
      >
        1
      </label>
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_1"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_1"
      >
        2
      </label>
    </li>
    <li
      className=""
      key="2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <input
        checked={false}
        id="checkbox_toggle_2"
        type="checkbox"
      />
      <label
        htmlFor="checkbox_toggle_2"
      >
        3
      </label>
    </li>
  </ul>
</div>
`;

exports[`Filter selector with single selection renders correctly with ariaLabel 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    aria-label="my label"
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className="filter-selected"
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      />
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        1
      </div>
    </li>
    <li
      className=""
      key="2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        2
      </div>
    </li>
    <li
      className=""
      key="3"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        3
      </div>
    </li>
  </ul>
</div>
`;

exports[`Filter selector with single selection renders correctly with emptyOptioncaption 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value="find me"
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className="filter-selected"
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        find me
      </div>
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        1
      </div>
    </li>
    <li
      className=""
      key="2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        2
      </div>
    </li>
    <li
      className=""
      key="3"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        3
      </div>
    </li>
  </ul>
</div>
`;

exports[`Filter selector with single selection renders correctly with no options 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className="filter-selected"
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      />
    </li>
  </ul>
</div>
`;

exports[`Filter selector with single selection renders correctly with options 1`] = `
<div
  className="dropdown-container"
  data-focusindex={0}
>
  <input
    aria-controls="undefined-dropdown-list"
    aria-expanded={false}
    aria-haspopup={true}
    className="form-control dropdown-triggerer"
    onClick={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ul
    className="dropdown-list false"
    data-focusindex={0}
    id="undefined-dropdown-list"
    role="menu"
    style={
      Object {
        "width": 0,
      }
    }
  >
    <li
      className="filter-selected"
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      />
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        1
      </div>
    </li>
    <li
      className=""
      key="2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        2
      </div>
    </li>
    <li
      className=""
      key="3"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="filter-label"
      >
        3
      </div>
    </li>
  </ul>
</div>
`;
